site/docs/contribute/environment.md

```diff
@@ -46,3 +46,36 @@ a docker registry as a proxy for all public images. This registry runs in a dock
 export DOCKERHUB_USERNAME='your docker hub username'
 export DOCKERHUB_PASSWORD='your password'
 ```
+
+## Optional Code Improvement Tools 
+## pre-commit
+
```

Comment:
you can add a pre-commit validation hook, that inspects the snapshot that's about to be committed according to the configured `golanci-lint`.  Its configuration is in `.pre-commit-config`

(Instead of lines 53- 55).